code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -25,33 +25,122 @@
 	faction = list("neutral")
 	attack_same = TRUE
 	gold_core_spawnable = HOSTILE_SPAWN
-	var/icon_resting = "goose_sit"
+	var/random_retaliate = TRUE
+	var/icon_vomit_start = "vomit_start"
+	var/icon_vomit = "vomit"
+	var/icon_vomit_end = "vomit_end"
 
+/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+	. = ..()
+	if(prob(5) && random_retaliate == TRUE)
+		Retaliate()
```

Comment:
Does it even have a valid set of targets?

Reply:
yes, that was in there before i did anything to the code